include/openvswitch/vlog.h

```diff
@@ -251,9 +251,14 @@ void vlog_rate_limit(const struct vlog_module *, enum vlog_level,
         case 'v':                               \
             vlog_set_verbosity(optarg);         \
             break;                              \
-        case OPT_LOG_FILE:                      \
-            vlog_set_log_file(optarg);          \
+        case OPT_LOG_FILE: {                    \
+            int err = vlog_set_log_file(optarg);\
+            /* exit if logfile explicitly given \
+             * but cannot be opened. */         \
+            if (err != 0 && optarg)             \
+                exit(EXIT_FAILURE);             \
```

Comment:
`vlog_set_log_file()` and `vlog_set_syslog_method()` handle errors they cannot recover from by logging a fatal error with `ovs_fatal()`, which in turn terminates the process. Perhaps `vlog_set_log_file()` should be doing the same thing to have consistent fatal error handling across functions from the same family.